postgis/lwgeom_ogc.c

```diff
@@ -848,14 +848,20 @@ Datum LWGEOM_asText(PG_FUNCTION_ARGS)
 	char *wkt;
 	size_t wkt_size;
 	text *result;
+	int dbl_dig_for_wkt = DBL_DIG;
 
 	POSTGIS_DEBUG(2, "Called.");
 
 	geom = PG_GETARG_GSERIALIZED_P(0);
 	lwgeom = lwgeom_from_gserialized(geom);
 
+	if (PG_NARGS() > 1)
+	{
+		dbl_dig_for_wkt = PG_GETARG_INT32(1);
+	}
+
```

Comment:
Continuing on with the idea from https://trac.osgeo.org/postgis/ticket/407, a default precision could be placed in the "else" part here by retrieving the `extra_float_digits` global, and something linke `dbl_dig_for_wkt = DBL_DIG + extra_float_digits;` or `dbl_dig_for_wkt += extra_float_digits`

This is based on the [PostgreSQL docs](http://www.postgresql.org/docs/current/static/runtime-config-client.html)
> The parameter value [extra_float_digits] is added to the standard number of digits (`FLT_DIG` or `DBL_DIG` as appropriate).